src/background/host.ts

```diff
@@ -34,23 +36,148 @@ export class Host {
+     */
+    private handleFriendMessage(message: IFriendMessage): void {
+        const numFriendsBefore = this.#friends.size;
+        // List of ids used for cleaning up friends list after updates
+        const newListIds: string[] = [];
+
+        for (const messageFriend of message.friends) {
+            const existingFriend = this.#friends.get(messageFriend.id);
+            const existingFriendPeerJsConnection = existingFriend?.connection;
+
+            // Message will contain the current user id
+            if (messageFriend.id !== this.id) {
+                const newFriendInstance = new Friend(messageFriend.id, messageFriend.displayName, existingFriendPeerJsConnection);
+                this.#friends.set(messageFriend.id, newFriendInstance);
+                newListIds.push(messageFriend.id);
+            }
+        }
+
+        // We've set the new friends, but we need to clean up any friends that weren't in the message
+        this.#friends.forEach(friend => {
+            if (newListIds.indexOf(friend.id) < 0) {
+                // Friend entry wasn't in the list of friends we got from the host
+                this.#friends.delete(friend.id);
+            }
+        });
+
+        const numFriendsAfter = this.#friends.size;
+        if (!this.hasNotifiedPartySize) {
+            // We now know the starting party size so notify the current user
+            this.notifyFriendChanges(this.#friends.size);
+        } else if (numFriendsBefore !== numFriendsAfter) {
+            // If the number of friends changed, notify the current user
+            const delta = numFriendsAfter - numFriendsBefore;
+            this.notifyFriendChanges(delta);
+        }
+    }
+
+    /**
+     * Notifies subscribers of a change in the friends list
+     * @param delta The change in amount of friends in the party
+     */
+    private notifyFriendChanges(delta: number): void {
+        this.hasNotifiedPartySize = true;
+        for (const cb of this.friendChangeSubscribers) {
+            cb(this.#friends.size, delta);
+        }
+    }
+
+    /**
+     * Method to handle when a peer connects to the current user if the current user is the host
+     * @param conn The peer connection
+     */
+    private onPeerConnectToHost(conn: Peer.DataConnection): void {
+        this.handleConnect(conn);
+
+        // Wait to send messages until peer has finished setting up their connection
+        setTimeout(() => {
+            // Notify everyone in the party with the updated friendslist
+            if (this.id == null) {
+                return;
+            }
+
+            const currentUser: IFriend = { id: this.id, displayName: this.displayName };
+            const friends = [];
+            this.#friends.forEach(friend => {
+                friends.push(friend.toSerializable());
+            });
+
+            // Include the current user is the friend message so peers include the host in their friend list
+            friends.push(currentUser);
```

Comment:
Or...

```ts
const friends = [
  ...this.#friends.map( ... ),
  currentUser
]
```

Reply:
yeah there's no .map :/

Reply:
uuuggghhh right `Set()` doesn't have `.map()`

Reply:
entries is an iterable so can't do that either haha